### PR TITLE
travis: build against 1.7

### PR DESCRIPTION
**Pull request by broady**
Test against 1.7beta2 to make sure nothing needs updating before the 1.7 release candidate and release.

### CODE REVIEW

**Comment by broady** [comment]
This looks like it could be a real failure in gRPC in Go 1.7

```
--- FAIL: TestTimeoutOnDeadServer (5.31s)
	end2end_test.go:420: Running test in tcp-clear environment...
	end2end_test.go:420: Running test in tcp-tls environment...
	end2end_test.go:420: Running test in unix-clear environment...
	end2end_test.go:420: Running test in unix-tls environment...
	end2end_test.go:420: Running test in handler-tls environment...
	end2end_test.go:1956: Leaked goroutine: goroutine 1 [chan receive]:
		testing.(*T).Run(0xc4200920c0, 0x86e7eb, 0x17, 0x89f6d0, 0xc420175d20)
			/home/travis/.gimme/versions/go1.7beta2.linux.amd64/src/testing/testing.go:647 +0x316
		testing.RunTests.func1(0xc4200920c0)
			/home/travis/.gimme/versions/go1.7beta2.linux.amd64/src/testing/testing.go:793 +0x6d
		testing.tRunner(0xc4200920c0, 0xc420041e30)
			/home/travis/.gimme/versions/go1.7beta2.linux.amd64/src/testing/testing.go:610 +0x81
		testing.RunTests(0x89f8a8, 0xa65b40, 0x20, 0x20, 0xc420175f20)
			/home/travis/.gimme/versions/go1.7beta2.linux.amd64/src/testing/testing.go:799 +0x2f0
		testing.(*M).Run(0xc420041ef8, 0x4)
			/home/travis/.gimme/versions/go1.7beta2.linux.amd64/src/testing/testing.go:743 +0x85
		main.main()
			google.golang.org/grpc/test/_test/_testmain.go:116 +0xc2
```

**Comment by iamqizhao** [comment]
I think you only need to let leak checker skip this goroutine as we did in https://github.com/grpc/grpc-go/blob/master/test/end2end_test.go#L1910.

**Comment by broady** [comment]
@iamqizhao PTAL. Tests pass locally.

**Comment by menghanl** [comment]
Travis failed on go vet. Should be fixed in #839.

**Comment by broady** [comment]
Updated with your patch, @menghanl and also to disable the composite literal vet check.

It fails on protoc generated code.

```
$ go tool vet -all .
reflection/grpc_testing/proto2.pb.go:28: github.com/golang/protobuf/proto.ExtensionRange composite literal uses unkeyed fields
```

**Comment by menghanl** [comment]
The go vet error on generated code cannot be fixed on our side...
An issue was already filed golang/protobuf#214

**Comment by broady** [comment]
It can. We turn the vet check off.

File: .travis.yml [hunk 1]
```diff
@@ -17,5 +18,5 @@ script:
   - '! gofmt -s -d -l . 2>&1 | read'
   - '! goimports -l . | read'
   - '! golint ./... | grep -vE "(_string|\.pb)\.go:"'
-  - '! go tool vet -all . 2>&1 | grep -vE "constant [0-9]+ not a string in call to Errorf"'
+  - '! go tool vet -all -composites=false . 2>&1 | grep -vE "constant [0-9]+ not a string in call to Errorf"'
```
Comment (tamird) [inline .travis.yml]: a "gentler" way to achieve this is to append ` | grep -vF 'pb.go:'` to the end of this line.

**Comment by tamird** [comment]
Should be closed now that #840 is merged.

**Pull request closed without merging**